aten/src/ATen/native/cuda/AdaptiveAveragePooling.cu

```diff
@@ -460,13 +461,16 @@ namespace {
 
         int osizeH = output_size[0];
         int osizeW = output_size[1];
-
         // preserve channels_last stride on output tensor;
         if (!output.is_contiguous(at::MemoryFormat::ChannelsLast)) {
           // TODO: modify this after resize_ added `memory_format` tag
           output.resize_({sizeB, sizeC, osizeH, osizeW}).as_strided_({sizeB, sizeC, osizeH, osizeW}, {sizeC*osizeH*osizeW, 1, osizeW*sizeC, sizeC});
         }
 
+        if (output.numel() == 0) {
+          return;
+        }
+
         const int max_threads = std::min<int>(
```

Comment:
*beginner question*
Why was this missing from the previous version? It seems like a fairly critical check no?

Reply:
The previous version was designed to error out when the number of elements in the `output` are 0. Since this one is supposed to return a valid output, we use the if-check.

Reply:
ah, makes sense. Thanks